### PR TITLE
Document Fusion compatibility for Oracle storage

### PR DESCRIPTION
**Pull request by pditommaso**
This PR document the Fusion compatibility with Oracle Object storage

### CODE REVIEW

File: docs/fusion.md [hunk 1]
```diff
@@ -274,6 +274,50 @@ The option `fusion.exportStorageCredentials` leaks the AWS credentials on the ta
...
+fusion.enabled = true
+fusion.exportStorageCredentials = true
+wave.enabled = true
+tower.accessToken = '<your platform access token>' // optional
+```
+
+Then you can run your pipeline using the following command:
+
+```bash
+nextflow run <YOUR PIPELINE> -work-dir s3://<YOUR BUCKET>/scratch
```
Comment (jordeu) [inline docs/fusion.md:305]:
```suggestion
nextflow run <YOUR_PIPELINE> -work-dir s3://<YOUR_BUCKET>/scratch
```

File: docs/fusion.md [hunk 2]
```diff
@@ -274,6 +274,50 @@ The option `fusion.exportStorageCredentials` leaks the AWS credentials on the ta
...
+```
+
+Replace `<YOUR PIPELINE>` and `<YOUR BUCKET>` with a pipeline script and bucket or your choice, for example:
```
Comment (jordeu) [inline docs/fusion.md:308]:
```suggestion
Replace `<YOUR_PIPELINE>` and `<YOUR_BUCKET>` with a pipeline script and bucket or your choice, for example:
```

File: docs/fusion.md [hunk 3]
```diff
@@ -274,6 +274,50 @@ The option `fusion.exportStorageCredentials` leaks the AWS credentials on the ta
...
+
+```bash
+nextflow run https://github.com/nextflow-io/rnaseq-nf -work-dir s3://nextflow-ci/scratch
+```
+
+Also, replace the region `ca-toronto-1` with the one matching your Oracle deployment.
```
Comment (jordeu) [inline docs/fusion.md:314]:
```suggestion
Also, replace `YOUR_ACCESS_KEY` / `YOUR_SECRET_KEY` with your [Customer Secret Key](https://docs.oracle.com/en-us/iaas/Content/Identity/Tasks/managingcredentials.htm#Working2). And `YOUR_BUCKET_NAMESPACE` / `YOUR_REGION` with the namespace and region of your bucket.
```

File: docs/fusion.md [hunk 4]
```diff
@@ -274,6 +274,50 @@ The option `fusion.exportStorageCredentials` leaks the AWS credentials on the ta
 This option should only be used for development purposes.
 :::
 
+### Local execution with Oracle Object Storage
+
+Fusion file system and Nextflow can be used with [Oracle Object Storage](https://www.oracle.com/es/cloud/storage/object-storage/)
```
Comment (jordeu) [inline docs/fusion.md:279]:
```suggestion
Fusion file system and Nextflow can be used with [Oracle Object Storage](https://www.oracle.com/cloud/storage/object-storage/)
```

File: docs/fusion.md [hunk 5]
```diff
@@ -274,6 +274,50 @@ The option `fusion.exportStorageCredentials` leaks the AWS credentials on the ta
...
+relying on the S3-like API compatibility provided by Oracle storage.
+
+This configuration requires the use of Docker (or similar container engine) for the execution of your pipeline tasks.
```
Comment (christopher-hakkaart) [inline docs/fusion.md:282]:
```suggestion
This configuration requires the execution of your pipeline tasks using Docker or a similar container engine.
```

File: docs/fusion.md [hunk 6]
```diff
@@ -274,6 +274,50 @@ The option `fusion.exportStorageCredentials` leaks the AWS credentials on the ta
 This option should only be used for development purposes.
 :::
 
+### Local execution with Oracle Object Storage
+
+Fusion file system and Nextflow can be used with [Oracle Object Storage](https://www.oracle.com/es/cloud/storage/object-storage/)
+relying on the S3-like API compatibility provided by Oracle storage.
```
Comment (christopher-hakkaart) [inline docs/fusion.md:280]:
```suggestion
They rely on the S3-like API compatibility provided by Oracle storage.
```

File: docs/fusion.md [hunk 7]
```diff
@@ -274,6 +274,50 @@ The option `fusion.exportStorageCredentials` leaks the AWS credentials on the ta
...
+
+This configuration requires the use of Docker (or similar container engine) for the execution of your pipeline tasks.
+
+The following configuration should be added in your Nextflow configuration file:
```
Comment (christopher-hakkaart) [inline docs/fusion.md:284]:
```suggestion
The following should be included in your Nextflow configuration file:
```

File: docs/fusion.md [hunk 8]
```diff
@@ -274,6 +274,50 @@ The option `fusion.exportStorageCredentials` leaks the AWS credentials on the ta
...
+
+```bash
+nextflow run https://github.com/nextflow-io/rnaseq-nf -work-dir s3://nextflow-ci/scratch
+```
+
+Also, replace the region `ca-toronto-1` with the one matching your Oracle deployment.
+
+:::{warning}
+The option `fusion.exportStorageCredentials` leaks the Oracle credentials on the task launcher script created by Nextflow.
+This option should only be used for development purposes.
```
Comment (christopher-hakkaart) [inline docs/fusion.md:318]:
```suggestion
The `fusion.exportStorageCredentials` option leaks the Oracle credentials to the Nextflow task launcher script and should only be used for development purposes.
```

**Review by christopher-hakkaart — CHANGES_REQUESTED** [review]
Added some small language suggestions. Please check for meaning.

File: docs/fusion.md [hunk 9]
```diff
@@ -267,13 +267,50 @@ Then you can run your pipeline using the following command:
...
+```bash
+nextflow run <YOUR_PIPELINE> -work-dir s3://<YOUR_BUCKET>/scratch
+```
+
+
+In the above snippet replace `<YOUR_ACCESS_KEY>` / `<YOUR_SECRET_KEY>` with your [Customer Secret Key](https://docs.oracle.com/en-us/iaas/Content/Identity/Tasks/managingcredentials.htm#Working2) and `<YOUR_BUCKET_NAMESPACE>` / `<YOUR_REGION>` with the namespace and region of your bucket.
+
+:::{warning}
+The `fusion.exportStorageCredentials` option leaks the Oracle credentials to the Nextflow task launcher script and should only be used for development purposes.
+
```
Comment (bentsherman) [inline docs/fusion.md:313]:
```suggestion
:::

```

**Pull request merged**